### PR TITLE
Deprecation audit and spring cleaning

### PR DESCRIPTION
**Pull request by samgreen**

### CODE REVIEW

File: HelloMixpanel/HelloMixpanel/SecretTableViewCell.m [hunk 1]
```diff
@@ -22,6 +22,7 @@ - (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSStr
 - (void)awakeFromNib
 {
     // Initialization code
+    [super awakeFromNib];
 }
```
Comment (yarneo) [inline HelloMixpanel/HelloMixpanel/SecretTableViewCell.m]: do we need this method?

File: HelloMixpanel/HelloMixpanelTests/HelloMixpanelTests.m [hunk 2]
```diff
@@ -904,18 +907,18 @@ - (void)testPeopleClearCharges
...
+//    for (NSInteger i = 0; i < 5005; i++) {
+//        [self.mixpanel track:@"rapid_event" properties:@{@"i": @(i)}];
+//    }
+//    [self waitForSerialQueue];
+//    XCTAssertTrue([self.mixpanel.eventsQueue count] == 5000);
+//    NSDictionary *e = self.mixpanel.eventsQueue[0];
+//    XCTAssertEqualObjects(e[@"properties"][@"i"], @(5));
+//    e = [self.mixpanel.eventsQueue lastObject];
+//    XCTAssertEqualObjects(e[@"properties"][@"i"], @(5004));
+//}
```
Comment (yarneo) [inline HelloMixpanel/HelloMixpanelTests/HelloMixpanelTests.m]: Is this meant to be removed as a test? If so, let's remove it

File: HelloMixpanel/Podfile [hunk 3]
```diff
@@ -1,10 +1,8 @@
 target :HelloMixpanelTests do
-    link_with 'HelloMixpanelTests'
     pod 'CocoaHTTPServer', '~> 2.2.1'
 end
 
 target :ABTestingTests do
-    link_with 'ABTestingTests'
     pod 'CocoaHTTPServer', '~> 2.2.1'
 end
```
Comment (yarneo) [inline HelloMixpanel/Podfile]: YES. Cocoapods 1.0!

File: Mixpanel.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -1242,39 +1234,36 @@
...
 				CLANG_WARN_ENUM_CONVERSION = YES;
 				CLANG_WARN_INT_CONVERSION = YES;
 				CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF = YES;
 				CLANG_WARN_OBJC_ROOT_CLASS = YES_ERROR;
 				CLANG_WARN_UNREACHABLE_CODE = YES;
-				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
 				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
 				COPY_PHASE_STRIP = YES;
-				CURRENT_PROJECT_VERSION = 1;
+				CURRENT_PROJECT_VERSION = 3.0.0;
```
Comment (yarneo) [inline Mixpanel.xcodeproj/project.pbxproj]: Could we remove #define VERSION @"2.9.9" in Mixpanel.m and get the info from this instead? I think we need the version for adding extra superproperties to tracks

File: Mixpanel/Mixpanel.m [hunk 5]
```diff
@@ -325,17 +319,11 @@ - (id)JSONSerializableObjectForObject:(id)obj
...
-    NSString *b64String = @"";
     NSData *data = [self JSONSerializeObject:array];
     if (data) {
-        b64String = [data mp_base64EncodedString];
-        b64String = CFBridgingRelease(CFURLCreateStringByAddingPercentEscapes(kCFAllocatorDefault,
-                                                                (__bridge CFStringRef)b64String,
-                                                                NULL,
-                                                                CFSTR("!*'();:@&=+$,/?%#[]"),
-                                                                kCFStringEncodingUTF8));
+        return MPURLEncode([data mp_base64EncodedString]);
```
Comment (yarneo) [inline Mixpanel/Mixpanel.m]: Have a look at: [data base64EncodedStringWithOptions:NSDataBase64Encoding64CharacterLineLength];
It was introduced in iOS7, and then we can get rid of the entire file NSData+MPBase64

File: HelloMixpanel/EventBindingTests/EventBindingTests.m [hunk 6]
```diff
@@ -165,7 +173,7 @@ - (void)testUIControlBindings
     XCTAssertEqual([selector selectFromRoot:rootViewController][0], c2, @"c2 should be selected by path");
 
     // Create binding and check state
-    MPUIControlBinding *binding = [MPEventBinding bindngWithJSONObject:eventParams];
+    MPUIControlBinding *binding = [MPEventBinding bindingWithJSONObject:eventParams];
```
Comment (yarneo) [inline HelloMixpanel/EventBindingTests/EventBindingTests.m]: Are we getting rid of: + (MPEventBinding *)bindngWithJSONObject:(NSDictionary *)object
I noticed it's deprecated and does the same thing as bindingWithJSONObject

File: HelloMixpanel/HelloMixpanel/SecretTableViewCell.m (hunk 1 above)
Comment (samgreen) [inline HelloMixpanel/HelloMixpanel/SecretTableViewCell.m]: It's a compile time error to forget to call `super` now

File: HelloMixpanel/EventBindingTests/EventBindingTests.m (hunk 6 above)
Comment (samgreen) [inline HelloMixpanel/EventBindingTests/EventBindingTests.m]: Yes, we deprecated months ago, removing now.

File: Mixpanel/Mixpanel.m (hunk 5 above)
Comment (samgreen) [inline Mixpanel/Mixpanel.m]: This is done in the networking PR.

File: Mixpanel.xcodeproj/project.pbxproj (hunk 4 above)
Comment (samgreen) [inline Mixpanel.xcodeproj/project.pbxproj]: That's the long term plan.

File: HelloMixpanel/HelloMixpanelTests/HelloMixpanelTests.m (hunk 2 above)
Comment (samgreen) [inline HelloMixpanel/HelloMixpanelTests/HelloMixpanelTests.m]: This test takes about 30 minutes...

Removed in another PR.

**Pull request merged**